iac-extensions/cloudformation/src/test/java/org/sonar/iac/cloudformation/plugin/CfnLintImporterTest.java

```diff
@@ -27,22 +27,29 @@
 import org.junit.jupiter.api.extension.RegisterExtension;
 import org.junit.jupiter.params.ParameterizedTest;
 import org.junit.jupiter.params.provider.CsvSource;
+import org.mockito.Mockito;
 import org.sonar.api.batch.fs.internal.TestInputFileBuilder;
 import org.sonar.api.batch.sensor.internal.SensorContextTester;
 import org.sonar.api.batch.sensor.issue.ExternalIssue;
+import org.sonar.api.notifications.AnalysisWarnings;
 import org.sonar.api.rules.RuleType;
 import org.sonar.api.utils.log.LogTesterJUnit5;
 import org.sonar.api.utils.log.LoggerLevel;
 import org.sonar.iac.cloudformation.reports.CfnLintImporter;
 
 import static org.assertj.core.api.AssertionsForInterfaceTypes.assertThat;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.when;
 
 class CfnLintImporterTest {
 
   @RegisterExtension
   public LogTesterJUnit5 logTester = new LogTesterJUnit5();
   private SensorContextTester context;
 
+  private final AnalysisWarnings spyAnalysisWarnings = spy(AnalysisWarnings.class);
```

Comment:
Why create a spy, when you don't validate raised warning. You could use a mock and check raised warnings. As well as no interactions with the mock. See TerraformProvidersTest for insperation.